### ISSUE TITLE
Intermittent NullPointerException from PerFileLineCoverage.hasValidBranch(PerFileLineCoverage.java:131)

### ISSUE THREAD

**Issue by eoghanoh1** [jmockit/jmockit1#267]
Please provide the following information:

* Version of JMockit that was used: 
Both jmockit version 1.21 & 1.22

* Description of the problem or enhancement request:

Using both jmockit and jmockit-coverage. 

An issue occurs where a unit test run intermittently fails caused by a null pointer exception that is thrown from the jmockit coverage jar.

The following is the stack trace;

java.lang.NullPointerException
	at mockit.coverage.lines.PerFileLineCoverage.hasValidBranch(PerFileLineCoverage.java:131) ~[jmockit-coverage.jar:1.22]
	at mockit.coverage.TestRun.branchExecuted(TestRun.java:46) ~[jmockit-coverage.jar:1.22]

This all I can provide from the stack trace as the rest of the code is proprietary. Hopefully this is enough information.

Also another intermittent issue was noticed with the following stack trace;

java.lang.ArrayIndexOutOfBoundsException: 65
	at mockit.coverage.lines.PerFileLineCoverage.registerExecution(PerFileLineCoverage.java:118) ~[jmockit-coverage.jar:1.21]
	at mockit.coverage.TestRun.lineExecuted(TestRun.java:33) ~[jmockit-coverage.jar:1.21]

It sounds like an issue resolved already in the next release 1.23;
https://github.com/jmockit/jmockit1/issues/264

Could you confirm this is the case.

**Comment by rliesenfeld** [comment]
There is no relation with #264.

Just knowing where each exception occurs is not enough, I am afraid. I would need some sample code which can reproduce the failures.

**Comment by eoghanoh1** [comment]
Is there any logging that can be enabled in jmockit for debug of the issue? As I mentioned this is intermittent and seems to happen 1 in 8/10 test runs. Could you give me some information on what exactly would you need to debug? Thanks.

**Comment by rliesenfeld** [comment]
No, no logging.

**Comment by eoghanoh1** [comment]
I would appreciate some information on what you need to debug this (as I asked for above).

**Comment by laurenfb** [comment]
Hi, 4 years later, I am seeing a similar error. It's super intermittent < 1% of our test runs but causing flaky enough tests that I'm hoping to resolve it. @eoghanoh1 did you ever make progress? I've done a bunch of investigation on the thrift side so far, but the null pointer exception is definitely coming from jmockit.

```
2020-05-16 06:55:12,524 pool-3-thread-2 ERROR: org.apache.thrift.server.TThreadPoolServer - Error occurred during processing of message.
java.lang.NullPointerException: null
	at mockit.coverage.lines.PerFileLineCoverage.hasValidBranch(PerFileLineCoverage.java:131) ~[jmockit-1.24.jar:?]
	at mockit.coverage.TestRun.branchExecuted(TestRun.java:46) ~[jmockit-1.24.jar:?]
```

**Comment by eoghanoh1** [comment]
Hi,
There was no further progress made on my side unfortunately. I did not have time to reproduce with sample code that I could share - our test suite code could not be shared online. Also I have since moved off the project that the tests suite was created for. Sorry I can't be of help - it was a frustrating error at the time for myself also. I hope you have more luck figuring it out.
Regards,
Eoghan
